src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs

```diff
@@ -36,102 +32,116 @@ private static string[] ParseMultiValue(string value)
-                new HeaderInfo(HttpKnownHeaderNames.Pragma,             false,  false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.ProxyAuthenticate,  false,  false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.ProxyAuthorization, false,  false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.ProxyConnection,    true,   false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.Range,              true,   false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.Referer,            true,   false,  false,  s_singleParser),
-                new HeaderInfo(HttpKnownHeaderNames.RetryAfter,         false,  false,  false,  s_singleParser),
-                new HeaderInfo(HttpKnownHeaderNames.Server,             false,  false,  false,  s_singleParser),
-                new HeaderInfo(HttpKnownHeaderNames.SetCookie,          false,  false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.SetCookie2,         false,  false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.TE,                 false,  false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.Trailer,            false,  false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.TransferEncoding,   true,   true,   true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.Upgrade,            false,  false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.UserAgent,          true,   false,  false,  s_singleParser),
-                new HeaderInfo(HttpKnownHeaderNames.Via,                false,  false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.Vary,               false,  false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.Warning,            false,  false,  true,   s_multiParser),
-                new HeaderInfo(HttpKnownHeaderNames.WWWAuthenticate,    false,  true,   true,   s_singleParser)
-            };
+            int offset = 0;
+            while (offset < length && char.IsWhiteSpace(value[startIndex + offset]))
+            {
+                offset++;
+            }
+
+            int end = length - 1;
+            while (end >= offset && char.IsWhiteSpace(value[startIndex + end]))
+            {
+                end--;
+            }
+
+            int newLength = end - offset + 1;
+            if (newLength == 0)
+            {
+                return string.Empty;
+            }
 
-            s_headerHashTable = new Hashtable(InfoArray.Length * 2, CaseInsensitiveAscii.StaticInstance);
-            for (int i = 0; i < InfoArray.Length; i++)
+            int newStartIndex = startIndex + offset;
+            if (newStartIndex == 0 && newLength == value.Length)
```

Comment:
It could be replaced with `Debug.Assert` but this package has no dependency on `System.Diagnostics.Debug` package, should I add it?